### PR TITLE
update storage NOTICE.txt

### PR DESCRIPTION
**Pull request by Jinming-Hu**
Closes https://github.com/Azure/azure-sdk-for-cpp/issues/5115

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]
Not sure if I am fully qualified to say if this is ok or not, but it looks rather ok to me. I don't see the "license notice for XXXX" text, maybe it is normal. I recommend to get a review from @RickWinter who logged the original issue.

**Comment by antkmsft** [comment]
"Closes #5115"?

**Comment by Jinming-Hu** [comment]
Hi @RickWinter , I'd like to get your approval before hitting the merge button.

File: sdk/storage/azure-storage-blobs/NOTICE.txt
Comment (RickWinter) [inline sdk/storage/azure-storage-blobs/NOTICE.txt:31]: The SDKs include this language in all NOTICE files.  Is there a reason you want to remove it?

File: sdk/storage/azure-storage-blobs/NOTICE.txt
Comment (RickWinter) [inline sdk/storage/azure-storage-blobs/NOTICE.txt:417]: "Azure/..." and "Microsoft/..." packages are not considered 3rd party OSS.  We do not need to list them.  Did you see or receive direction from CELA stating otherwise?

File: sdk/storage/azure-storage-blobs/NOTICE.txt
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/NOTICE.txt:31]: @RickWinter The whole file is generated here.

image

I didn't make any changes to the generated file (I don't think I should).

File: sdk/storage/azure-storage-blobs/NOTICE.txt
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/NOTICE.txt:417]: Sure, I'll remove `azure/azure-c-shared-utility`, `azure/macro-utils-c`, `azure/umock-c` and `microsoft/wil`.

**Comment by Jinming-Hu** [comment]
@RickWinter I've updated the files. Can you take another look?

**Pull request merged**
